### PR TITLE
#6Fixed CowBreed Model and Validation Strategy

### PR DESCRIPTION
**Pull request by mamesareagan**
This pull request addresses several issues and improvements related to the CowBreed model and its validation strategy. Here's a summary of the changes:

**Model Fix:**

Corrected the BreedChoices enum to match the choices provided in the CowBreed model.
Resolved the AttributeError issue when trying to create a breed with an invalid name.
**Validation Strategy:**

Updated the CustomCowBreedValidator to use the correct ValidationError import from Django.
Adjusted the validation logic to ensure the breed name is within the specified choices and is not an empty string.
**Tests:**

Added new tests to cover various scenarios, including creating breeds with valid and invalid names, as well as empty names.
Fixed existing tests to align with the corrected model and validation strategy.

### CODE REVIEW

File: dairy/settings.py [hunk 1]
```diff
@@ -89,6 +89,10 @@
         "NAME": "django.contrib.auth.password_validation.NumericPasswordValidator",
     },
 ]
+# Add the custom validator for CowBreed model
```
Comment (peter-evance) [inline dairy/settings.py:92]: Unnecessary implementation, this file (settings.py) is solely for global settings for our whole application, model related stuff should be in the respective app folder

File: tests/users/tests/test_models.py [hunk 2]
```diff
@@ -54,3 +57,38 @@ def test_create_superuser(self):
         assert superuser.is_staff
         assert superuser.is_superuser
         assert superuser.is_active
+
+@pytest.mark.django_db
+class TestCowBreedModel:
```
Comment (peter-evance) [inline tests/users/tests/test_models.py:62]: Test case should be moved into another app test folder, this is for users app and it's related test logic

File: users/models.py [hunk 3]
```diff
@@ -156,3 +156,52 @@ def __str__(self):
     def save(self, *args, **kwargs):
         self.clean()
         super().save(*args, **kwargs)
+
+class CowBreed(models.Model):
```
Comment (peter-evance) [inline users/models.py:160]: Model should be moved to the core app, not users.models.py

File: users/urls.py [hunk 4]
```diff
@@ -10,11 +10,16 @@
 router = routers.DefaultRouter()
 router.register(r"users", CustomUserViewSet, basename="users")
 
+# Add a new router for CowBreed
```
Comment (peter-evance) [inline users/urls.py:13]: Clean up the code, remove unnecessary comments

File: users/validators.py [hunk 5]
```diff
@@ -52,4 +52,37 @@ def validate_username(cls, username):
             .exclude(username=username)
             .exists()
         ):
+            
             raise ValidationError("Username already exists.")
+
+class CustomCowBreedValidator:
```
Comment (peter-evance) [inline users/validators.py:58]: Should be moved to a much appropriate app location—core app

**Review by peter-evance — CHANGES_REQUESTED** [review]

File: users/models.py (hunk 3 above)
Comment (mamesareagan) [reply users/models.py]: What does this mean,I quite didn't get it clearly

Comment (peter-evance) [reply users/models.py]: > What does this mean,I quite didn't get it clearly

Models such as cow breed , cow, milk etc are the core of our application thus they should be in the `core` app , users app is/must only have code related to users management and roles to separate concerns and make the code base easy to understand

**Review by mamesareagan — COMMENTED** [review]
I have checked the issues raised and they are all fixed as expected.

**Pull request closed without merging**